# Airbnb Listings Exploratory Data Analysis

## Objective
Explore Airbnb listing data to understand what factors may influence listing prices, availability, and listing characteristics.

## Dataset
This dataset contains Airbnb listing information including price, location, room type, availability and review metrics.

In [33]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

pd.set_option("display.max_columns", None)
pd.set_option('display.float_format', '{:.2f}'.format) 

In [34]:
df = pd.read_csv("../data/listings.csv", encoding='utf-8-sig')

df.shape

(27945, 18)

The dataset contains 27945 rows and 18 columns. Each row represent Airbnb listing in Tokyo.

In [35]:
df.head()

,id,name,host_id,host_name,neighbourhood_group,neighbourhood,latitude,longitude,room_type,price,minimum_nights,number_of_reviews,last_review,reviews_per_month,calculated_host_listings_count,availability_365,number_of_reviews_ltm,license
0,197677,Oshiage Holiday Apartment,964081,Yoshimi & Marek,NaN,Sumida Ku,35.72,139.83,Entire home/apt,12600.00,3,190,2025-08-24,1.11,1,183,10,M130003350
1,776070,Kero-kero room 1F,801494,Kei,NaN,Kita Ku,35.74,139.77,Private room,10459.00,3,272,2025-05-21,1.74,1,76,14,M130000243
2,905944,4F Spacious Apartment in Shinjuku / Shibuya Tokyo,4847803,Best Stay In Tokyo!,NaN,Shibuya Ku,35.68,139.68,Entire home/apt,33671.00,5,281,2025-09-21,1.85,9,305,50,Hotels and Inns Business Act | 渋谷区保健所長 | 31渋健生...
3,1016831,5 mins Shibuya Cat modern sunny Shimokita,5596383,Wakana,NaN,Setagaya Ku,35.66,139.67,Private room,24143.00,1,284,2025-09-21,1.87,1,80,16,M130001107
4,1196177,Homestay at Host's House - Senju-Ohashi Station,5686404,Yukiko,NaN,Adachi Ku,35.74,139.80,Private room,8795.00,2,150,2025-09-07,1.01,1,253,25,M130007760


Missing datas in neighborhood_group\
What does calculated_host_listing and number_of_reviews_Itm mean?

In [36]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 27945 entries, 0 to 27944
Data columns (total 18 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              27945 non-null  int64  
 1   name                            27945 non-null  str    
 2   host_id                         27945 non-null  int64  
 3   host_name                       27938 non-null  str    
 4   neighbourhood_group             0 non-null      float64
 5   neighbourhood                   27945 non-null  str    
 6   latitude                        27945 non-null  float64
 7   longitude                       27945 non-null  float64
 8   room_type                       27945 non-null  str    
 9   price                           25480 non-null  float64
 10  minimum_nights                  27945 non-null  int64  
 11  number_of_reviews               27945 non-null  int64  
 12  last_review                     24009 non-n

Neighbourhood, room_type are categorical.\
Latitude, longitude, price, minimum_nights, number_of_reviews, reviews_per_month, availability_365 are numerical.\
id, name, host_id, host_namem last_review, license are possible text columns.\
Neighborhood_group doesnt have any data.\
Missing values in host_name, price, last_review, reviews_per_month and license.

In [37]:
df.describe(include="all")

,id,name,host_id,host_name,neighbourhood_group,neighbourhood,latitude,longitude,room_type,price,minimum_nights,number_of_reviews,last_review,reviews_per_month,calculated_host_listings_count,availability_365,number_of_reviews_ltm,license
count,27945.00,27945,27945.00,27938,0.00,27945,27945.00,27945.00,27945,25480.00,27945.00,27945.00,24009,24009.00,27945.00,27945.00,27945.00,27929
unique,NaN,24320,NaN,3740,NaN,49,NaN,NaN,4,NaN,NaN,NaN,978,NaN,NaN,NaN,NaN,17121
top,NaN,Sleep Inn Gotanda,NaN,圭太,NaN,Shinjuku Ku,NaN,NaN,Entire home/apt,NaN,NaN,NaN,2025-09-15,NaN,NaN,NaN,NaN,Hotels and Inns Business Act | 墨田区保健所 | 5墨福衛生環...
freq,NaN,40,NaN,3174,NaN,4991,NaN,NaN,23759,NaN,NaN,NaN,1003,NaN,NaN,NaN,NaN,68
mean,949854396596311552.00,NaN,378915257.98,NaN,NaN,NaN,35.70,139.74,NaN,24781.88,5.32,35.97,NaN,1.68,25.35,170.26,13.34,NaN
std,537995573532604224.00,NaN,217787784.38,NaN,NaN,NaN,0.04,0.07,NaN,77865.75,10.72,66.33,NaN,1.54,34.25,98.21,18.09,NaN
min,197677.00,NaN,322234.00,NaN,NaN,NaN,35.52,139.08,NaN,1500.00,1.00,0.00,NaN,0.01,1.00,0.00,0.00,NaN
25%,762438150388641408.00,NaN,189502014.00,NaN,NaN,NaN,35.69,139.70,NaN,11201.50,1.00,2.00,NaN,0.72,4.00,99.00,1.00,NaN
50%,1169462436136334080.00,NaN,425423543.00,NaN,NaN,NaN,35.70,139.73,NaN,16200.00,2.00,16.00,NaN,1.35,11.00,159.00,8.00,NaN
75%,1362098392910273792.00,NaN,557366491.00,NaN,NaN,NaN,35.72,139.79,NaN,25045.00,2.00,42.00,NaN,2.27,34.00,247.00,20.00,NaN


price range: 1500 - 9999998, average is 24781.88\
number of reviews range: 0 - 3705\
availability range: 0 - 365

Preliminary analysis to help us understand the data. (Could be some missing). 

### Key Variables

These columns are likely to be important for analysis:
- price : price of the listing per night.
- room_type : type of listing ( Entire home/apt, private room, shared room, hotel room).
- neighbourhood: the general area where the listing is located.
- availability_365 : how many days are the listing available in a year.
- number_of_reviews : how many reviews does the listing have.
- minimum_nights : minimum number of nights required to book the listing.

## Initial questions

1. Does room type influence price?
2. Which neighborhoods have the highest price?
3. Do listings with more reviews charge more?
4. How does availability vary across listings?
5. Does minimum nights have an effect on price?